--- lxqt-globalkeys-0.8.0/CMakeLists.txt.unify	2014-10-10 20:41:14.000000000 -0300
+++ lxqt-globalkeys-0.8.0/CMakeLists.txt	2014-12-19 19:20:03.678530145 -0200
@@ -8,6 +8,8 @@
 	set(CMAKE_BUILD_TYPE Release)
 endif()
 
+set(LIB_INSTALL_DIR "lib${LIB_SUFFIX}" CACHE PATH "Installation path for libraries")
+
 # Set default library installation path
 set(LIB_INSTALL_DIR "lib${LIB_SUFFIX}" CACHE PATH "Installation directory for libraries")
 
@@ -16,26 +18,29 @@
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
 set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 set(CMAKE_AUTOMOC ON)
+
+find_package(lxqt REQUIRED QUIET)
 if(USE_QT5)
     cmake_minimum_required(VERSION 2.8.11)
     find_package(Qt5Widgets REQUIRED)
     find_package(Qt5DBus REQUIRED)
     find_package(Qt5LinguistTools REQUIRED QUIET)
-    find_package(lxqt-qt5 REQUIRED QUIET)
-    set(QT_VERSION_SUFFIX "-qt5")
     message(STATUS "Building with Qt${Qt5Core_VERSION_STRING}")
 else()
     set(QT_USE_QTDBUS 1)
     find_package(Qt4 REQUIRED)
     include(${QT_USE_FILE})
     find_package(LXQT REQUIRED)
-    set(QT_VERSION_SUFFIX "")
     message(STATUS "Building with Qt${QTVERSION}")
 endif()
 include(${LXQT_USE_FILE})
 
-set(LXQT_GLOBALKEYS_LIBRARY_NAME lxqt-globalkeys${QT_VERSION_SUFFIX})
-set(LXQT_GLOBALKEYS_UI_LIBRARY_NAME lxqt-globalkeys-ui${QT_VERSION_SUFFIX})
+# Standard directories for installation
+include(GNUInstallDirs)
+
+set(LXQT_GLOBALKEYS_LIBRARY_NAME lxqt-globalkeys)
+set(LXQT_GLOBALKEYS_UI_LIBRARY_NAME lxqt-globalkeys-ui)
+set(LXQT_INSTALL_CMAKE_DIR     ${CMAKE_INSTALL_FULL_LIBDIR}/cmake)
 
 #************************************************
 # Build config.cmake files
@@ -74,11 +79,11 @@
 add_subdirectory(client)
 add_subdirectory(ui)
 
-install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${LXQT_GLOBALKEYS_CMAKE_NAME}-config.cmake DESTINATION share/cmake/${LXQT_GLOBALKEYS_CMAKE_NAME})
-install(FILES cmake/lxqt_globalkeys_use.cmake DESTINATION share/cmake/${LXQT_GLOBALKEYS_CMAKE_NAME})
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${LXQT_GLOBALKEYS_CMAKE_NAME}-config.cmake DESTINATION ${LXQT_INSTALL_CMAKE_DIR}/${LXQT_GLOBALKEYS_CMAKE_NAME})
+install(FILES cmake/lxqt_globalkeys_use.cmake DESTINATION ${LXQT_INSTALL_CMAKE_DIR}/${LXQT_GLOBALKEYS_CMAKE_NAME})
 
-install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${LXQT_GLOBALKEYS_UI_CMAKE_NAME}-config.cmake DESTINATION share/cmake/${LXQT_GLOBALKEYS_UI_CMAKE_NAME})
-install(FILES cmake/lxqt_globalkeys_ui_use.cmake DESTINATION share/cmake/${LXQT_GLOBALKEYS_UI_CMAKE_NAME})
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${LXQT_GLOBALKEYS_UI_CMAKE_NAME}-config.cmake DESTINATION ${LXQT_INSTALL_CMAKE_DIR}/${LXQT_GLOBALKEYS_UI_CMAKE_NAME})
+install(FILES cmake/lxqt_globalkeys_ui_use.cmake DESTINATION ${LXQT_INSTALL_CMAKE_DIR}/${LXQT_GLOBALKEYS_UI_CMAKE_NAME})
 
 # building tarball with CPack -------------------------------------------------
 include (InstallRequiredSystemLibraries)
--- lxqt-globalkeys-0.8.0/ui/CMakeLists.txt.unify	2014-12-19 19:21:16.915703890 -0200
+++ lxqt-globalkeys-0.8.0/ui/CMakeLists.txt	2014-12-19 19:21:39.428757299 -0200
@@ -7,14 +7,13 @@
 	set(QT_DBUS_PREFIX "org.qtproject")
 endif()
 
+set(LXQT_GLOBALKEYS_LIBRARY lxqt-globalkeys)
 if(USE_QT5)
     set(QTX_INCLUDE_DIRS "")
     set(QTX_LIBRARIES Qt5::Widgets Qt5::DBus)
-    set(LXQT_GLOBALKEYS_LIBRARY lxqt-globalkeys-qt5)
 else()
     set(QTX_INCLUDE_DIRS ${QT_QTCORE_INCLUDE_DIR} ${QT_QTGUI_INCLUDE_DIR} ${QT_QTDBUS_DIR})
     set(QTX_LIBRARIES ${QT_QTCORE_LIBRARY} ${QT_QTGUI_LIBRARY} ${QT_QTDBUS_LIBRARY})
-    set(LXQT_GLOBALKEYS_LIBRARY lxqt-globalkeys)
 endif()
 set(${PROJECT_NAME}_HEADER_NAMESPACE "LXQtGlobalKeysUi")
 
