--- /home/PisiLinux/docky-2.2.1/Docky/gtk-gui/gui.stetic
+++ Docky/gtk-gui/gui.stetic
@@ -1,16 +1,12 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
+<?xml version="1.0" encoding="utf-8"?>
 <stetic-interface>
   <configuration>
     <images-root-path>..</images-root-path>
     <target-gtk-version>2.12</target-gtk-version>
   </configuration>
   <import>
+    <widget-library name="notify-sharp, Version=0.4.0.0, Culture=neutral, PublicKeyToken=2df29c54e245917a" />
     <widget-library name="wnck-sharp, Version=2.20.0.0, Culture=neutral, PublicKeyToken=35e10195dab3c99f" />
-    <widget-library name="gnomedesktop-sharp, Version=2.20.0.0, Culture=neutral, PublicKeyToken=35e10195dab3c99f" />
-    <widget-library name="../../Docky.Services/bin/Debug/Docky.Services.dll" />
-    <widget-library name="../../Docky.Items/bin/Debug/Docky.Items.dll" />
-    <widget-library name="../../Docky.DBus/bin/Debug/Docky.DBus.dll" />
-    <widget-library name="../../Docky.Widgets/bin/Debug/Docky.Widgets.dll" />
     <widget-library name="../bin/Debug/Docky.exe" internal="true" />
   </import>
   <widget class="Gtk.Bin" id="Docky.Interface.DockPreferences" design-size="500 356">
@@ -309,7 +305,7 @@
       </widget>
     </child>
   </widget>
-  <widget class="Gtk.Window" id="Docky.ConfigurationWindow" design-size="380 390">
+  <widget class="Gtk.Window" id="Docky.ConfigurationWindow" design-size="398 390">
     <property name="MemberName" />
     <property name="Title" translatable="yes">Docky Settings</property>
     <property name="Icon">stock:gtk-preferences Menu</property>
