--- lxqt-policykit-0.8.0.orig/CMakeLists.txt	2014-06-28 14:21:37.000000000 -0300
+++ lxqt-policykit-0.8.0/CMakeLists.txt	2014-11-10 15:11:12.878904005 -0200
@@ -7,6 +7,8 @@
 # Support different versions of Qt
 option(USE_QT5 "Build with Qt5." $ENV{USE_QT5})
 
+include(GNUInstallDirs)
+
 # find Qt binding of polkit
 if(USE_QT5)
     find_package(PolkitQt5-1) # Qt5 version
@@ -113,7 +115,7 @@
     )
 
     # install
-    install(TARGETS lxqt-policykit-agent DESTINATION bin)
+	install(TARGETS lxqt-policykit-agent DESTINATION ${CMAKE_INSTALL_FULL_LIBEXECDIR})
 
 endif()
 
