Index: lxqt-panel-0.7.0/plugin-mount/CMakeLists.txt
===================================================================
--- lxqt-panel-0.7.0.orig/plugin-mount/CMakeLists.txt
+++ lxqt-panel-0.7.0/plugin-mount/CMakeLists.txt
@@ -14,6 +14,7 @@ set(HEADERS
 )

 set(SOURCES
+    ${CMAKE_SOURCE_DIR}/panel/lxqtpanelpluginconfigdialog.cpp
     lxqtmountplugin.cpp
     mountbutton.cpp
     menudiskitem.cpp
@@ -26,6 +27,7 @@ set(SOURCES
 )

 set(MOCS
+    ${CMAKE_SOURCE_DIR}/panel/lxqtpanelpluginconfigdialog.h
     lxqtmountplugin.h
     mountbutton.h
     menudiskitem.h

Index: lxqt-panel-0.7.0/plugin-volume/CMakeLists.txt
===================================================================
--- lxqt-panel-0.7.0.orig/plugin-volume/CMakeLists.txt
+++ lxqt-panel-0.7.0/plugin-volume/CMakeLists.txt
@@ -13,6 +13,7 @@ set(HEADERS
 )

 set(SOURCES
+    ${CMAKE_SOURCE_DIR}/panel/lxqtpanelpluginconfigdialog.cpp
     lxqtvolume.cpp
     volumebutton.cpp
     volumepopup.cpp
@@ -22,6 +23,7 @@ set(SOURCES
 )

 set(MOCS
+    ${CMAKE_SOURCE_DIR}/panel/lxqtpanelpluginconfigdialog.h
     lxqtvolume.h
     volumebutton.h
     volumepopup.h

Index: lxqt-panel-0.7.0/plugin-worldclock/lxqtworldclock.h
===================================================================
--- lxqt-panel-0.7.0.orig/plugin-worldclock/lxqtworldclock.h
+++ lxqt-panel-0.7.0/plugin-worldclock/lxqtworldclock.h
@@ -66,7 +66,6 @@ public:
     QDialog *configureDialog();

 private slots:
-    void synchroTimeout();
     void timeout();
     void wheelScrolled(int);

Index: lxqt-panel-0.7.0/plugin-kbindicator/lxqtkbindicatorconfiguration.h
===================================================================
--- lxqt-panel-0.7.0.orig/plugin-kbindicator/lxqtkbindicatorconfiguration.h
+++ lxqt-panel-0.7.0/plugin-kbindicator/lxqtkbindicatorconfiguration.h
@@ -50,12 +50,6 @@ public:
 public slots:
     void saveSettings();

-    void selectColour(const QString &);
-    void on_typeCOB_currentIndexChanged(int);
-    void on_sourceCOB_currentIndexChanged(int);
-    void on_maximumHS_valueChanged(int);
-    void on_fontB_clicked();
-

 private:
     Ui::LxQtKbIndicatorConfiguration *ui;

Index: lxqt-panel-0.7.0/plugin-dom/domplugin.cpp
===================================================================
--- lxqt-panel-0.7.0.orig/plugin-dom/domplugin.cpp
+++ lxqt-panel-0.7.0/plugin-dom/domplugin.cpp
@@ -40,7 +40,6 @@ DomPlugin::DomPlugin(const ILxQtPanelPlu
     ILxQtPanelPlugin(startupInfo)
 {
     mButton.setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Expanding);
-    mButton.setIcon(XdgIcon::fromTheme("preferences-plugin"));
     connect(&mButton, SIGNAL(clicked()), this, SLOT(showDialog()));
 }

