--- lxqt-panel-0.8.0.git9f27751.old/plugin-mount/actions/deviceaction.h 2014-08-19 12:59:32.192343730 -0700
+++ lxqt-panel-0.8.0.git9f27751/plugin-mount/actions/deviceaction.h 2014-08-19 13:50:08.978761035 -0700
@@ -31,7 +31,7 @@

 #include <QObject>
 #include <QSettings>
-#include <LXQtMount/Mount>
+#include <lxqtmount/LXQtMount/Mount>

 namespace {
 class MountDevice;
diff -rupN lxqt-panel-0.8.0.git9f27751.old/plugin-mount/lxqtmountplugin.cpp lxqt-panel-0.8.0.git9f27751/plugin-mount/lxqtmountplugin.cpp
--- lxqt-panel-0.8.0.git9f27751.old/plugin-mount/lxqtmountplugin.cpp    2014-08-19 12:59:32.193343748 -0700
+++ lxqt-panel-0.8.0.git9f27751/plugin-mount/lxqtmountplugin.cpp    2014-08-19 13:47:26.336938097 -0700
@@ -28,7 +28,7 @@

 #include "lxqtmountplugin.h"
 #include "lxqtmountconfiguration.h"
-#include <LXQtMount/Mount>
+#include <lxqtmount/LXQtMount/Mount>
 #include "actions/deviceaction.h"
 #include "popup.h"
 #include "mountbutton.h"
diff -rupN lxqt-panel-0.8.0.git9f27751.old/plugin-mount/menudiskitem.cpp lxqt-panel-0.8.0.git9f27751/plugin-mount/menudiskitem.cpp
--- lxqt-panel-0.8.0.git9f27751.old/plugin-mount/menudiskitem.cpp   2014-08-19 12:59:32.193343748 -0700
+++ lxqt-panel-0.8.0.git9f27751/plugin-mount/menudiskitem.cpp   2014-08-19 13:54:02.997825478 -0700
@@ -27,7 +27,7 @@

 #include <QDesktopServices>
 #include "menudiskitem.h"
-#include <LXQtMount/Mount>
+#include <lxqtmount/LXQtMount/Mount>
 #include <XdgIcon>
 #include <QDebug>
 #include <QUrl>
diff -rupN lxqt-panel-0.8.0.git9f27751.old/plugin-mount/popup.cpp lxqt-panel-0.8.0.git9f27751/plugin-mount/popup.cpp
--- lxqt-panel-0.8.0.git9f27751.old/plugin-mount/popup.cpp  2014-08-19 12:59:32.193343748 -0700
+++ lxqt-panel-0.8.0.git9f27751/plugin-mount/popup.cpp  2014-08-19 13:56:22.289246053 -0700
@@ -29,7 +29,7 @@
 #include <QGridLayout>
 #include <QLabel>
 #include <QDesktopWidget>
-#include <LXQtMount/Mount>
+#include <lxqtmount/LXQtMount/Mount>
 #include "menudiskitem.h"
 #include "popup.h"

