diff -up lxqt-common-0.8.0/autostart/CMakeLists.txt.libexec lxqt-common-0.8.0/autostart/CMakeLists.txt
diff -up lxqt-common-0.8.0/autostart/lxqt-policykit-agent.desktop.in.libexec lxqt-common-0.8.0/autostart/lxqt-policykit-agent.desktop.in
--- lxqt-common-0.8.0/autostart/lxqt-policykit-agent.desktop.in.libexec	2014-08-23 04:12:54.000000000 -0500
+++ lxqt-common-0.8.0/autostart/lxqt-policykit-agent.desktop.in	2014-12-19 07:39:50.814981890 -0600
@@ -1,8 +1,8 @@
 [Desktop Entry]
 Type=Application
 Name=PolicyKit Handler
-TryExec=lxqt-policykit-agent
-Exec=lxqt-policykit-agent
+TryExec=/usr/libexec/lxqt-policykit-agent
+Exec=/usr/libexec/lxqt-policykit-agent
 OnlyShowIn=LXQt;
 X-LXQt-Module=true
 
