--- configure.ac	2015-02-10 02:00:40.000000000 +0200
+++ configure.ac	2015-03-01 03:38:27.590663888 +0200
@@ -75,16 +75,16 @@
 AM_PROG_VALAC([0.26], [], [AC_MSG_ERROR([valac is required to build Budgie Desktop])])
 
 # Determine if upower is newer than we're supporting right now
-PKG_CHECK_MODULES(UPOWER99, [upower-glib >= 0.9.99], [have_upower0999=yes], [have_upower0999=no])
-if test "x$have_upower0999" = "xyes" ; then
-    VALAFLAGS="$VALAFLAGS --define HAVE_UPOWER0999"
+PKG_CHECK_MODULES(UPOWER93, [upower-glib >= 0.93.23], [have_upower093=yes], [have_upower093=no])
+if test "x$have_upower093" = "xyes" ; then
+    VALAFLAGS="$VALAFLAGS --define HAVE_UPOWER093"
 fi
 
 
-# Check for GTK3.13+ (Le World Blow Up)
-PKG_CHECK_MODULES(GTK313, [gtk+-3.0  >= 3.13], [have_gtk313=yes], [have_gtk313=no])
-if test "x$have_gtk313" = "xyes" ; then
-    VALAFLAGS="$VALAFLAGS --define HAVE_GTK313"
+# Check for GTK3.12+ (Le World Blow Up)
+PKG_CHECK_MODULES(GTK312, [gtk+-3.0  >= 3.12], [have_gtk312=yes], [have_gtk312=no])
+if test "x$have_gtk312" = "xyes" ; then
+    VALAFLAGS="$VALAFLAGS --define HAVE_GTK312"
 fi
 
 # Check for glib >= 2.40 (GAppInfoMonitor introduction)
